### PR TITLE
Fix rocksmq load with prefix

### LINKED ISSUE #7620: Standalone oom when recover from existing storage.
#### Steps/Code to reproduce:
1. Run standalone and run pytest for some times (I ran it for about 3 hours)
2. Stop standalone
3. Start standalone with the same rocksmq path.

#### Expected result:
Start standalone successfully.

#### Actual results:
System memory usage is rising very fast and causes OOM eventually.

#### Environment:
- Milvus version(05464377):
- Deployment mode(standalone):
- SDK version(2.0.0rc6.dev3):
- OS(Ubuntu): 
- CPU/Memory:  Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz   64GB
- GPU: 
- Others: 

#### Additional context:
The last goroutines runs before OOM listed as follows (sampled with pprof):

```
goroutine profile: total 54
9 @ 0x793dbb 0x131715c 0x1332999 0x132540b 0x1342dfb 0x1344332 0x802fe1
#	0x131715b	github.com/tecbot/gorocksdb._Cfunc_rocksdb_iter_next+0x3b							_cgo_gotypes.go:1867
#	0x1332998	github.com/tecbot/gorocksdb.(*Iterator).Next.func1+0x58								
/home/godchen/go/pkg/mod/github.com/tecbot/gorocksdb@v0.0.0-20191217155057-f0fad39f321c/iterator.go:79
#	0x132540a	github.com/tecbot/gorocksdb.(*Iterator).Next+0x2a								
/home/godchen/go/pkg/mod/github.com/tecbot/gorocksdb@v0.0.0-20191217155057-f0fad39f321c/iterator.go:79
#	0x1342dfa	github.com/milvus-io/milvus/internal/util/rocksmq/server/rocksmq.prefixLoad+0x1fa				
/home/godchen/Documents/milvus/internal/util/rocksmq/server/rocksmq/rocksmq_retention.go:76
#	0x1344331	github.com/milvus-io/milvus/internal/util/rocksmq/server/rocksmq.(*retentionInfo).loadRetentionInfo+0xbf1	
/home/godchen/Documents/milvus/internal/util/rocksmq/server/rocksmq/rocksmq_retention.go:192

8 @ 0x7e583b 0x1342fa8 0x1344332 0x802fe1
#	0x1342fa7	github.com/milvus-io/milvus/internal/util/rocksmq/server/rocksmq.prefixLoad+0x3a7				
/home/godchen/Documents/milvus/internal/util/rocksmq/server/rocksmq/rocksmq_retention.go:80
#	0x1344331	github.com/milvus-io/milvus/internal/util/rocksmq/server/rocksmq.(*retentionInfo).loadRetentionInfo+0xbf1	
/home/godchen/Documents/milvus/internal/util/rocksmq/server/rocksmq/rocksmq_retention.go:192

4 @ 0x793dbb 0x13170a9 0x1332865 0x1325269 0x1342e6a 0x1344332 0x802fe1
#	0x13170a8	github.com/tecbot/gorocksdb._Cfunc_rocksdb_iter_key+0x48							_cgo_gotypes.go:1853
#	0x1332864	github.com/tecbot/gorocksdb.(*Iterator).Key.func1+0x64								
/home/godchen/go/pkg/mod/github.com/tecbot/gorocksdb@v0.0.0-20191217155057-f0fad39f321c/iterator.go:60
#	0x1325268	github.com/tecbot/gorocksdb.(*Iterator).Key+0x48								
/home/godchen/go/pkg/mod/github.com/tecbot/gorocksdb@v0.0.0-20191217155057-f0fad39f321c/iterator.go:60
#	0x1342e69	github.com/milvus-io/milvus/internal/util/rocksmq/server/rocksmq.prefixLoad+0x269				
/home/godchen/Documents/milvus/internal/util/rocksmq/server/rocksmq/rocksmq_retention.go:77
#	0x1344331	github.com/milvus-io/milvus/internal/util/rocksmq/server/rocksmq.(*retentionInfo).loadRetentionInfo+0xbf1	
/home/godchen/Documents/milvus/internal/util/rocksmq/server/rocksmq/rocksmq_retention.go:192
```

### PR DESCRIPTION
**Pull request by fishpenguin**
Signed-off-by: fishpenguin <kun.yu@zilliz.com>

Resolves: #7620

### CODE REVIEW

File: internal/util/rocksmq/server/rocksmq/rocksmq_retention.go [hunk 1]
```diff
@@ -76,10 +76,10 @@ func prefixLoad(db *gorocksdb.DB, prefix string) ([]string, []string, error) {
 	for ; iter.Valid(); iter.Next() {
 		key := iter.Key()
 		value := iter.Value()
-		defer key.Free()
-		defer value.Free()
 		keys = append(keys, string(key.Data()))
+		key.Free()
 		values = append(values, string(value.Data()))
```
Comment (czs007) [inline internal/util/rocksmq/server/rocksmq/rocksmq_retention.go:81]: Is there also a suspicion of accessing dangling pointers here?
refer to https://github.com/milvus-io/milvus/issues/7300

Comment (fishpenguin) [reply internal/util/rocksmq/server/rocksmq/rocksmq_retention.go:81]: NO!

Comment (czs007) [reply internal/util/rocksmq/server/rocksmq/rocksmq_retention.go:81]: why? Is string conversion guaranteed to copy data?

**Pull request merged**